NightscoutServiceKit/OTPManager.swift

```diff
@@ -74,41 +74,49 @@ public class OTPManager {
     let issuerName = "Loop"
     var tokenPeriod: TimeInterval
     var passwordDigitCount = 6
+    let maxOTPsToAccept: Int
+    
     public static var defaultTokenPeriod: TimeInterval = 30
+    public static var defaultMaxOTPsToAccept = 2
     
-    public init(secretStore: OTPSecretStore = KeychainManager(), nowDateSource: @escaping () -> Date = {Date()}, tokenPeriod: TimeInterval = OTPManager.defaultTokenPeriod) {
+    public init(secretStore: OTPSecretStore = KeychainManager(), nowDateSource: @escaping () -> Date = {Date()}, tokenPeriod: TimeInterval = OTPManager.defaultTokenPeriod, maxOTPsToAccept: Int = OTPManager.defaultMaxOTPsToAccept) {
         self.secretStore = secretStore
         self.nowDateSource = nowDateSource
         self.tokenPeriod = tokenPeriod
+        self.maxOTPsToAccept = maxOTPsToAccept
         if secretStore.tokenSecretKey() == nil || secretStore.tokenSecretKeyName() == nil {
             resetSecretKey()
         }
     }
     
-    public func validateOTP(otpToValidate: String) -> Bool {
-        let maxOTPsToAccept = 2
+    public func validateOTP(otpToValidate: String) throws {
         
-        guard otpToValidate.count == passwordDigitCount && getLastPasswordsAscending(count: maxOTPsToAccept).contains(otpToValidate) else {
-            return false //Doesn't match
+        guard otpToValidate.count == passwordDigitCount else {
+            throw OTPValidationError.invalidFormat(otp: otpToValidate)
         }
         
```

Comment:
We can't assume we only store the last 2 passwords anymore so this needed changed to be more flexible.